presto-main/src/test/java/io/prestosql/sql/planner/TestDynamicFilter.java

```diff
@@ -112,13 +111,53 @@ public void testJoinOnCast()
     {
         assertPlan("SELECT o.orderkey FROM orders o, lineitem l WHERE cast(l.orderkey as int) = cast(o.orderkey as int)",
                 anyTree(
-                        node(
-                                JoinNode.class,
-                                anyTree(
+                        join(
+                                INNER,
+                                ImmutableList.of(equiJoinClause("expr_orders", "expr_lineitem")),
+                                ImmutableMap.of(),
```

Comment:
now that we use this version of `join` matcher, we could still use `anyTree`. Otherwise this test is more fragile (e.g if some plan nodes are added by other rules independently)